src/write_outputs/write_net_revenue.jl

```diff
@@ -32,6 +32,16 @@ function write_net_revenue(path::AbstractString, inputs::Dict, setup::Dict, EP::
 		dfNetRevenue.Var_OM_cost_out *= ModelScalingFactor # converting Million US$ to US$
 	end
 
+	# Add capacity constraint dual
+	dfNetRevenue[!, "Capacity_constraint_dual"] = fill(NaN, nrow(dfNetRevenue))
+	for y in intersect(dfGen[dfGen.Max_Cap_MW.>0, :R_ID], 1:G)
+		dual_val = dual.(EP[:cMaxCap][y])
+		if setup["ParameterScale"] == 1
+			dual_val *= ModelScalingFactor # converting MM$/GW-yr to $/MW-yr
+		end
+		dfNetRevenue[y, "Capacity_constraint_dual"] = dual_val
```

Comment:
Consider separating the `for` and `if`. Could you `*=` the column afterward?

Or look at the solutions in other files: above the for loop have `modelscalingfactor = setup["ParameterScale"] ? ModelScalingFactor : 1` and then always `* modelscalingfactor`.

Reply:
Is there a reason why the intersection is necessary? `R_ID` is always in the set 1:G.

Reply:
Yes it needs to be only the generators with a maximum capacity constraint.